app/src/main/java/org/kiwix/kiwixmobile/data/local/KiwixDatabase.java

```diff
@@ -69,83 +72,87 @@ public String getName() {
         BookDatabaseEntity.TABLE,
         LibraryDatabaseEntity.TABLE,
         RecentSearch.TABLE,
-        Bookmarks.TABLE,
+        Bookmark.TABLE,
         NetworkLanguageDatabaseEntity.TABLE,
         History.TABLE
     };
   }
 
   @Override
   protected boolean onUpgrade(SQLiteDatabaseWrapper db, int oldVersion, int newVersion) {
-    if (newVersion >= 3 && oldVersion < 3) {
-      db.execSQL("DROP TABLE IF EXISTS recents");
-      tryCreateTable(RecentSearch.TABLE);
-    }
-    if (newVersion >= 3 && (oldVersion < 3 || oldVersion == 7 || oldVersion == 6)) {
-      db.execSQL("DROP TABLE IF EXISTS recents");
-      db.execSQL("DROP TABLE IF EXISTS recentsearches");
-      tryCreateTable(RecentSearch.TABLE);
-    }
-    if (newVersion >= 3) {
-      tryCreateTable(RecentSearch.TABLE);
-    }
-    if (newVersion >= 4) {
-      tryCreateTable(Bookmarks.TABLE);
-    }
-    if (newVersion >= 5 && oldVersion < 5) {
-      db.execSQL("DROP TABLE IF EXISTS book");
-      tryCreateTable(BookDatabaseEntity.TABLE);
-    }
-    if (newVersion >= 5) {
-      tryCreateTable(BookDatabaseEntity.TABLE);
-    }
-    if (newVersion >= 6 && oldVersion < 6) {
-      db.execSQL("DROP TABLE IF EXISTS Bookmarks");
-      tryCreateTable(Bookmarks.TABLE);
-      migrateBookmarks();
-    }
-    if (newVersion >= 6) {
-      tryCreateTable(Bookmarks.TABLE);
-    }
-    if (newVersion >= 9) {
```

Comment:
@mhutti1 The problem was because of these type of **if** statements with no check on the **oldVersion**.